### PR TITLE
Filter api call form

### PR DESCRIPTION
**Pull request by tuna42na**
Some updates on the form filter functionality.

### CODE REVIEW

File: src/components/NameListContext.js [hunk 1]
```diff
@@ -1,17 +1,108 @@
...
+    const callNames = async () => {
+      const names = await axios(listURL);
+      this.setState({ list: names.data, listView: names.data });
+      this.onChangeNumber(this.state.namesPerPage);
+    };
+    callNames();
+  };
+
+  // Ordering Names Switch Function
+  onChangeOrder = async (value) => {
```
Comment (trvslhlt) [inline src/components/NameListContext.js:56]: Why is this an async function?

File: src/components/NameListContext.js [hunk 2]
```diff
@@ -1,17 +1,108 @@
...
+    if (value == "name") {
+      const results = [...this.state.list].sort((a, b) => {
+        let varA = a.name.toUpperCase();
```
Comment (trvslhlt) [inline src/components/NameListContext.js:59]: Use meaningful variable names when you can. Possibly `firstName` and `lastName` here.

File: src/components/NameListContext.js [hunk 3]
```diff
@@ -1,17 +1,108 @@
...
+      callList: this.callList,
+    };
+  }
+
+  componentDidUpdate() {}
+
+  // Call the List from the Server => Sets List Component
+  callList = (listURL) => {
+    const callNames = async () => {
+      const names = await axios(listURL);
```
Comment (trvslhlt) [inline src/components/NameListContext.js]: What happens if this request fails?

File: src/components/NameListContext.js (hunk 2 above)
Comment (tuna42na) [reply src/components/NameListContext.js:59]: I agree, but this is only for first names...

File: src/components/NameListContext.js (hunk 3 above)
Comment (tuna42na) [reply src/components/NameListContext.js]: I do need to make a catch here. I'll update.

File: src/components/NameListContext.js (hunk 1 above)
Comment (tuna42na) [reply src/components/NameListContext.js:56]: If the function is not async the setState({}) will not wait until the results are set and the page will crash. Everything will be referencing a blank array of objects.

File: src/components/NameListContext.js (hunk 3 above)
Comment (tuna42na) [reply src/components/NameListContext.js]: Updated.

**Comment by cwkarwisch** [comment]
No comments from me.

**Pull request merged**